backends/p4tools/modules/testgen/testgen.cpp

```diff
@@ -115,6 +115,14 @@ int Testgen::mainImpl(const IR::P4Program *program) {
     if (seed) {
         printFeature("test_info", 4, "============ Program seed %1% =============\n", *seed);
     }
+    if (testgenOptions.minCoverage > 0 && !testgenOptions.coverageOptions.coverageEnabled()) {
```

Comment:
This is something I would catch in the options early, if possible.

Reply:
Options should be processed as early as possible to get immediate feedback. Otherwise, you run your tests for a couple of minutes then get this error. 

We typically validate options already for each lambda we add. For example, here you could throw an error if you want to enable minCoverage without previously enable coverage in some way. This way you also do not require an extra validate call. This requires ordering, but that seems fine to me. This is a limitation of the way our options work.

Reply:
I don't like requiring one to be set before another, that is why I did not want to put the validation into the lambdas.